server/server.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/allegro/bigcache"
+)
+
+const (
+	apiVersion  = "v1"
+	apiBasePath = "/api/" + apiVersion + "/"
+
+	// path to cache.
+	cachePath = apiBasePath + "cache/"
+)
+
+var (
+	shards             int
+	maxEntriesInWindow int
+	lifeWindow         int64
+	hardMax            int
+	verbose            bool
+	maxEntrySize       int
+	logfile            string
+	port               int
+	cache              *bigcache.BigCache
+)
+
+func init() {
+	flag.BoolVar(&verbose, "v", false, "Verbose logging.")
+	flag.IntVar(&shards, "shards", 1024, "Number of shards for the cache.")
+	flag.IntVar(&maxEntriesInWindow, "maxInWindow", 1000*10*60, "Used only in initial memory allocation.")
+	flag.Int64Var(&lifeWindow, "lifetime", 10, "Lifetime, in minutes, of each cache object.")
+	flag.IntVar(&hardMax, "max", 8192, "Maximum amount of objects in the cache.")
+	flag.IntVar(&maxEntrySize, "maxShardEntrySize", 500, "The maximum size of each object stored in a shard. Used only in initial memory allocation.")
+	flag.IntVar(&port, "port", 9090, "The port to listen on.")
+	flag.StringVar(&logfile, "logfile", "", "Location of the logfile.")
+}
+
+func main() {
+	flag.Parse()
+
+	var logger *log.Logger
+
+	if logfile == "" {
+		logger = log.New(os.Stdout, "", log.LstdFlags)
+	} else {
+		f, err := os.OpenFile(logfile, os.O_APPEND|os.O_WRONLY, 0600)
+		if err != nil {
+			panic(err)
+		}
+		logger = log.New(f, "", log.LstdFlags)
+	}
+
+	config := bigcache.Config{
+		Shards:             shards,
+		LifeWindow:         time.Duration(lifeWindow) * time.Minute,
+		MaxEntriesInWindow: maxEntriesInWindow,
+		MaxEntrySize:       maxEntrySize,
+		Verbose:            verbose,
+		HardMaxCacheSize:   hardMax,
+		OnRemove:           nil,
+		Logger:             logger,
+	}
```

Comment:
this could be removed with global cache config.

Reply:
Fixed.